### PR TITLE
Dev update 20231215

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin, please rerender

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin, please rerender

**Comment by Tobias-Fischer** [comment]
Needs https://github.com/conda-forge/colmap-feedstock/pull/30

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin, please rerender

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin, please rerender

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin  please rerender

**Comment by carterbox** [comment]
My general tips are to use the latest CMake CUDA features such as `enable_language(CUDA)` and/or `find_package(CUDAToolkit)` (not `find_package(CUDA)`).

Specifically for this recipe, it doesn't seem like having the CUDA compiler is necessary at all, since [there doesn't seem to be any device code in pycolmap](https://github.com/search?q=repo%3Acolmap%2Fpycolmap+COLMAP_CUDA_ENABLED&type=code), so I'm not sure why your CMakeLists are searching for the CUDA Toolkit. Probably, you can just build one cuda/noncuda variant for each python version because the ABI for all of the CUDA variants of colmap is the same. An example of this is [astra-toolbox](https://github.com/conda-forge/astra-toolbox-feedstock/blob/main/recipe/meta.yaml), where the CUDA variant of this python package links to any of the CUDA builds of libastra.

**Comment by Tobias-Fischer** [comment]
Thanks @carterbox! Looks like all is green now. I think upstream will soon combine colmap and pycolmap in a single repo anyhow, so I won’t be changing / patching here further.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -51,6 +53,8 @@ requirements:
   host:
     - pip
     - python
+    - scikit-build
+    - scikit-build-core
```
Comment (jakirkham) [inline recipe/meta.yaml:57]: Is there a reason to add both `scikit-build` & `scikit-build-core`?

Thought it was one or the other. IIUC `scikit-build-core` is [replacing `scikit-build`]( https://scikit-build-core.readthedocs.io/en/latest/#scikit-build-core )

**Pull request merged**